src/status_im/ui/screens/chat/message/message.cljs

```diff
@@ -616,6 +616,73 @@
                 [message.audio/message-content message] [message-status message]]]]]
             reaction-picker])))
 
+(defn contact-request-status-pending []
+  [react/view {:style {:flex-direction :row}}
+   [quo/text {:style  {:margin-right 5.27}
+              :weight :medium
+              :color :secondary}
+    (i18n/label :t/contact-request-pending)]
+   [react/activity-indicator {:animating true
+                              :size      :small
+                              :color     colors/gray}]])
+
+(defn contact-request-status-accepted []
+  [quo/text {:style  {:color colors/green}
+             :weight :medium}
+   (i18n/label :t/contact-request-accepted)])
+
+(defn contact-request-status-declined []
+  [quo/text {:style  {:color colors/red}
+             :weight :medium}
+   (i18n/label :t/contact-request-declined)])
+
+(defn contact-request-status-label [state]
+  [react/view {:style {:width 136
+                       :border-radius 8
+                       :flex 1
+                       :justify-content :center
+                       :align-items :center
+                       :background-color (when (= :retry state)
+                                           colors/blue-light)
+                       :border-width 1
+                       :border-color (case state
+                                       constants/contact-request-state-accepted colors/green-transparent-10
+                                       constants/contact-request-state-declined colors/red-light
+                                       constants/contact-request-state-pending colors/gray-lighter)
+                       :padding-vertical 10
+                       :padding-horizontal 16}}
+
+   (case state
+     constants/contact-request-state-pending  [contact-request-status-pending]
+     constants/contact-request-state-accepted [contact-request-status-accepted]
+     constants/contact-request-state-declined [contact-request-status-declined])])
+
+(defmethod ->message constants/content-type-contact-request
+  [{:keys [outgoing] :as message} _]
+  [react/view {:style {:width 168
+                       :min-height 224.71
+                       :border-radius 8
+                       :border-width 1
+                       :border-color colors/gray-lighter
+                       :align-items :center
+                       :padding-bottom 10
+                       :margin-vertical 4
+                       :align-self (if outgoing :flex-end :flex-start)
+                       :margin-right (if outgoing 8 0)
+                       :margin-left (if outgoing 0 8)}}
+   [react/image {:source (resources/get-image :hand-wave)
+                 :style  {:width 112
+                          :height 96.71}}]
```

Comment:
that pretty weird number, why it's not round ?

Reply:
Not sure :) but the UI will have to be changed when we redesign (using still old designs), but I'll make it round